controller/src/main/java/io/pravega/controller/server/rpc/grpc/GuardianAuthorizer.java

```diff
@@ -0,0 +1,48 @@
+package io.pravega.controller.server.rpc.grpc;
+
+import com.emc.nautilus.auth.client.GuardianClient;
+import com.emc.nautilus.auth.client.GuardianClientFactory;
+import com.emc.nautilus.auth.models.acls.TargetedACLs;
+import com.google.common.base.Strings;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class GuardianAuthorizer implements ServerInterceptor {
+    private final GuardianClientFactory guardianClientFactory;
+    private final ConcurrentMap<String, GuardianClient> clientMap;
+
+    public GuardianAuthorizer(String guardianIp) {
+        this.guardianClientFactory = new GuardianClientFactory(guardianIp);
+        clientMap = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+        String method = headers.get(Metadata.Key.of("method", Metadata.ASCII_STRING_MARSHALLER));
+
+        if (!Strings.isNullOrEmpty(method) && method.equals("guardian")) {
+            String userName = headers.get(Metadata.Key.of("userName", Metadata.ASCII_STRING_MARSHALLER));
+            String token = headers.get(Metadata.Key.of("token", Metadata.ASCII_STRING_MARSHALLER));
+
+            GuardianClient client;
+            synchronized (this) {
+                if (!clientMap.containsKey(token)) {
+                    client = guardianClientFactory.withToken(token);
+                    clientMap.putIfAbsent(token, client);
+                } else {
+                    client = clientMap.get(token);
+                }
+            }
+            // TODO: check ACLs for the resource string
+            //String resourcePath = "";
+            //TargetedACLs acls = client.getUserTargetedACLsForResourcePath(resourcePath);
```

Comment:
Looking ahead, the call you'll want here is actually "lookup" (https://asdstash.isus.emc.com/projects/NAUT/repos/platform/browse/guardian-client/src/main/java/com/emc/nautilus/auth/client/GuardianClient.java#347)

The difference is that `getUserTargetedACLsForResourcePath` is a "static" get on acls that are assigned directly to all users on that resource.   (it's the kind of call the UI would make to see what's explicitly assigned on something, to display it on a grid and allow edits.).  --> This call is only allowed by PLATFORM_ADMINs.

The "lookup" call will retrieve a full list of all the global roles and targeted roles, for this user only, grouped by target (resource).  It will also factor in acls that this user may have received through group membership (ldap).   --> this call can be called by anybody (it uses "your own token" to figure out what you have access to)